lib/activerecord-mysql-structure/active_record/structure_sql_sanitizer.rb

```diff
@@ -153,5 +184,36 @@ def sort_indices(lines)
 
       sorted_lines
     end
+
+    def sort_versions(lines)
+      versions = []
+      sorted_lines = []
+      in_versions = false
+      lines.each do |line|
+        if in_versions
+          if MIGRATION_VERSION_VALUE_REGEX.match(line)
+            # Collect the version values until we run out (generally due to the newlines at the end)
```

Comment:
Will this fail if an extra new line is missing?

Reply:
If the dump process somehow produces a schema with no trailing newlines, you're correct, no output would be produced. I glossed over this possibility primarily because we presently have the opposite problem (AR adds two trailing newlines instead of just one), but easy enough to extract this to a private method that we invoke both here and at the end if `in_versions` never became false again.